<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <group ns="/openArm">    
        <!-- Robot model -->
        <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find arm_description)/urdf/Arm_URDF.xacro'" />
        <arg name="x" default="0"/>
        <arg name="y" default="0"/>
        <arg name="z" default="0"/>
        
        <!-- Spawn the robot model -->
        <node name="mybot_spawn" pkg="gazebo_ros" type="spawn_model" output="screen"
              args="-urdf -param robot_description -model arm -x $(arg x) -y $(arg y) -z $(arg z)" />
    
        <!-- Load Controllers-->
        <rosparam command="load" file="$(find arm_description)/config/joints.yaml"/>

        <!-- Controllers -->
        <node name="controller_spawner" pkg="controller_manager" type="spawner"
            respawn="false" output="screen" ns="/openArm"
            args="--namespace=/openArm
            joint_state_controller
            joint1_position_controller
            joint2_position_controller
            joint3_position_controller
            joint4_position_controller
            --timeout 60">
        </node>

    </group>      
</launch>